squashfs-tools/xz_wrapper.c

```diff
@@ -53,6 +56,11 @@ static float dictionary_percent = 0;
  * Two specific options are supported:
  *	-Xbcj
  *	-Xdict-size
+ *	-Xpreset
+ *	-Xe
```

Comment:
Making `extreme` a separate option makes sense. I'm a bit confused why you have this here in the comment but not actually implementing the idea.

Reply:
this was the original implementation but once i actually started working on this, i decided to put it in preset.

extreme is actually a preset and putting that in a separate option would put some confusion since -Xpreset would be ignored. (I could be wrong, but I remember at times and I discovered at the very end the extreme flag overwrite any compression level and just apply maximum level + some extra option. But happy to analyze this further and change code to have this feature merged!)

So for parsing simplicity (to skip validating or checking -Xpreset) i just made a change and now we have `-Xpreset extreme`.

Reply:
I'm not that familiar with the internals of `xz` either, but if you look at this part, you can see that `level` and `extreme` flag are orthogonal for the most part.

https://github.com/xz-mirror/xz/blob/e7da44d5151e21f153925781ad29334ae0786101/src/liblzma/lzma/lzma_encoder_presets.c#L51

`squashfs-tools-ng` is also a great source of reference since this is already implemented there.

Reply:
@wychen yep i analyzed xz source and i verified that those are flags and adds stuff to the compression level (flags are OR to the compression level value)

I updated this pr with the additional handling.